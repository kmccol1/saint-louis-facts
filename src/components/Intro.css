/* General styling for the Intro section */
.intro {
  background: linear-gradient(135deg, #e3f2fd, #bbdefb);
  padding: 3rem 2rem;
  text-align: center;
  border-radius: 12px;
  box-shadow: 0 6px 20px rgba(0, 0, 0, 0.1);
  margin: 2rem auto;
  max-width: 900px;
  font-family: 'Roboto', sans-serif;
  transition: transform 0.3s ease, box-shadow 0.3s ease;
}

.intro:hover {
  transform: translateY(-5px);
  box-shadow: 0 10px 25px rgba(0, 0, 0, 0.15);
}

/* Title styling for the Intro section */
.intro h2 {
  font-size: 2.8rem;
  font-weight: 700;
  color: #1a237e;
  margin-bottom: 1rem;
  text-transform: capitalize;
  letter-spacing: 1.2px;
  font-family: 'Montserrat', sans-serif;
}

/* Paragraph styling */
.intro p {
  font-size: 1.2rem;
  line-height: 1.8;
  color: #37474f;
  margin: 0 auto;
  max-width: 750px;
  font-family: 'Open Sans', sans-serif;
}

/* Mobile responsiveness */
@media (max-width: 768px) {
  .intro {
    padding: 2rem 1.5rem;
  }

  .intro h2 {
    font-size: 2.2rem;
  }

  .intro p {
    font-size: 1rem;
  }
}
