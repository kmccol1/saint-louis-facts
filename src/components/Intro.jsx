import React from 'react';
import './Intro.css'; // Import the CSS file for styling

function Intro()
{
  return (
    <section className="intro" aria-label="Introduction">
    <h2>Discover the Heart of Saint Louis</h2>
      <p>
        Nestled along the mighty Mississippi, Saint Louis is a city that thrives on its vibrant spirit, storied past, and welcoming community.
        From the iconic Gateway Arch standing tall as a beacon of progress to its soulful blues, barbecue, and neighborhoods brimming with charm,
        this city invites you to celebrate its unique blend of history and modern energy. Dive into these carefully curated facts and uncover
        the essence of the Gateway to the West!
      </p>
    </section>
  );
}

export default Intro;
