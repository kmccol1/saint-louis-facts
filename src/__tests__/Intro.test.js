import { render, screen } from '@testing-library/react';
import Intro from '../components/Intro';

test('renders intro section with heading and text', () => {
  render(<Intro />);
  expect(screen.getByText(/Why Saint Louis?/i)).toBeInTheDocument();
  expect(screen.getByText(/Saint Louis, Missouri, is a city rich in history/i)).toBeInTheDocument();
});
