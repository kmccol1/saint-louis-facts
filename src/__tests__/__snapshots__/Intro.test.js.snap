// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro Component renders the intro section with correct structure 1`] = `
<div>
  <section
    aria-label="Introduction"
    class="intro"
  >
    <h2>
      Discover the Heart of Saint Louis
    </h2>
    <p>
      Nestled along the mighty Mississippi, Saint Louis is a city that thrives on its vibrant spirit, storied past, and welcoming community. From the iconic Gateway Arch standing tall as a beacon of progress to its soulful blues, barbecue, and neighborhoods brimming with charm, this city invites you to celebrate its unique blend of history and modern energy. Dive into these carefully curated facts and uncover the essence of the Gateway to the West!
    </p>
  </section>
</div>
`;
